### ISSUE TITLE
Conversion of sfe Object to AnnData

### ISSUE THREAD

**Issue by aasingh2** [pachterlab/SpatialFeatureExperiment#56]
Hello,

I wanted to ask if there is a built-in function to convert a sfe object to AnnData object, or if there is a way to save the sfe object as a h5ad file to facilitate interoperability with Python.

Thanks!

**Comment by alikhuseynov** [comment]
I don't think it is planned right now, but maybe @lambdamoses has some ideas.
FYI, there is [VoyagerPy]( https://github.com/pachterlab/voyagerpy/tree/dev2 ) and updated python workflow for [Xenium](https://github.com/pachterlab/voyagerpy/blob/dev2/examples/xenium.ipynb)
There will hopefully be integration with SpatialData https://github.com/pachterlab/voyagerpy/issues/29

**Comment by lambdamoses** [comment]
For now you can use some package that converts between SCE and AnnData such as zellkonverter because SFE inherits on SCE, and then deal with the spatial part separately in a manner that is consistent with squidpy. I think it would be a valuable new feature to incorporate. 

Also, the [ArtifactDB](https://github.com/ArtifactDB) project has the alabaster packages for language agnostic serialization of Bioc objects from R; the Python counterpart are the dolomite packages. I just submitted alabaster.sfe to Bioc recently; there's no dolomite for SFE yet, but I should we should write it.

**Comment by vraghu90** [comment]
On the same lines, I have an AnnData object which I would like to convert to spatial feature experiment.  Is there a workflow to do this and and run the Moran's I calculation?

**Comment by lambdamoses** [comment]
Not yet, but I think there should be. For now, you can convert AnnData to SCE with something like zellkonverter first and then add the geometries and coordinates. Please consider opening a pull request.

**Comment by lambdamoses** [comment]
Also, you don't need to convert AnnData to SFE to compute Moran' I. Squidpy can do that, so can voyagerpy: https://github.com/pachterlab/voyagerpy/tree/dev2 Although at present only the R implementation of Voyager has the bivariate and multivariate spatial statistics.

**Comment by vraghu90** [comment]
Can we do a heatmap of the moran's I scores using voyagerpy?

Best,
Vineeth

---

*Vineeth Raghuram, Ph.D.*

*(603)-943-0489*

**Comment by lambdamoses** [comment]
What kind of heatmap? Do you mean plotting local Moran's I's at spatial locations or global Moran's I across genes and samples?

**Comment by lambdamoses** [comment]
You can plot local Moran's I with the `plotLocalResult` function in Voyager (I think it's snake case in the Python implementation) but you need to do some data wrangling yourself to plot Moran's I across genes and samples. In the R implementation, global spatial results for gene expression are stored in `rowData` and the column names indicate `sample_id`. `rowData` is an S4 `DataFrame`; you can extract the relevant columns and convert that into a matrix which can be plotted as a heatmap with the `pheatmap` package. I'm not sure how well the Python implementation of Voyager handles different samples though.

**Comment by lambdamoses** [comment]
Any more questions? If not then I'll close the issue.

**Comment by vraghu90** [comment]
Quick question on this. Is there a workflow where I can convert a previously analyzed/processed seurat object(Xenium) into a SFE to perform Voyager operations?

**Comment by alikhuseynov** [comment]
> Quick question on this. Is there a workflow where I can convert a previously analyzed/processed seurat object(Xenium) into a SFE to perform Voyager operations?

Yes, take a look here -> https://pachterlab.github.io/SpatialFeatureExperiment/dev/articles/SFE.html#coercion-from-seurat
